x/claim/keeper/claim.go

```diff
@@ -0,0 +1,235 @@
+package keeper
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/arkeonetwork/arkeo/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/gogo/protobuf/proto"
+)
+
+// SetClaimRecord sets a claim record for an address in store
+func (k Keeper) SetClaimRecord(ctx sdk.Context, claimRecord types.ClaimRecord) error {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, chainToStorePrefix(claimRecord.Chain))
+
+	bz, err := proto.Marshal(&claimRecord)
+	if err != nil {
+		return err
+	}
+
+	addr := []byte(strings.ToLower(claimRecord.Address))
+
+	// validate address if valid based on chain
+	switch claimRecord.Chain {
+	case types.ETHEREUM:
+		if !types.IsValidEthAddress(claimRecord.Address) {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid eth address (%s)", err)
+		}
+	case types.ARKEO:
+		_, err := sdk.AccAddressFromBech32(claimRecord.Address)
+		if err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid arkeo address (%s)", err)
+		}
+	case types.THORCHAIN:
+		return errors.New("thorchain not supported yet")
+	}
+
+	prefixStore.Set(addr, bz)
+	return nil
+}
+
+// SetClaimables set claimable amount from balances object
+func (k Keeper) SetClaimRecords(ctx sdk.Context, claimRecords []types.ClaimRecord) error {
+	for _, claimRecord := range claimRecords {
+		err := k.SetClaimRecord(ctx, claimRecord)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetClaimables get claimables for genesis export
+func (k Keeper) GetClaimRecords(ctx sdk.Context, chain types.Chain) []types.ClaimRecord {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, chainToStorePrefix(chain))
+
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	claimRecords := []types.ClaimRecord{}
+	for ; iterator.Valid(); iterator.Next() {
+
+		claimRecord := types.ClaimRecord{}
+
+		err := proto.Unmarshal(iterator.Value(), &claimRecord)
+		if err != nil {
+			panic(err)
```

Comment:
don't panic , try to return the error , and bubble it up to some place that can be dealt with

Reply:
[b8e6a19](https://github.com/arkeonetwork/arkeo/pull/29/commits/b8e6a19a0ac8a0b1fae4365cb6aecfb9b9fe628b)